### ISSUE TITLE
App crashes at Hotline initialization while running a background task

### ISSUE THREAD

**Issue by pramod636** [freshdesk/hotline-ios#15]
We have the Hotline initialization code in didFinishLaunching which gets called during the execution of the background tasks as well and it is here that the Hotline SDK initialization crashes. Since only the library (.a) files have been shipped, the crash stack stops at the point where we call the init. We are digging into the issue ourselves. But since it has caused more than 400 crashes for 300 odd users, we are reporting it quick. Please let us know if you need any further information from our end.

Crash stack

``` 
0x100576998 specialized AppDelegate.application(UIApplication, didFinishLaunchingWithOptions : [UIApplicationLaunchOptionsKey : Any]?) -> Bool (AppDelegate.swift:56)
0x100571938 @objc AppDelegate.application(UIApplication, didFinishLaunchingWithOptions : [UIApplicationLaunchOptionsKey : Any]?) -> Bool (AppDelegate.swift)
```

Line 56 has the initialization code.

**Comment by hrishikesh-p** [comment]
Hi @pramod-practo 

We tried reproducing this issue using the clues you have provided with no luck. The initialisation was working fine , in background fetch mode, in background task, and other threads. 

If you use crashlytics, Can you please email the crash link (shareable one) to support@hotline.io ? 

Also , can you list the Holtine APIs you are invoking on launch and during backgroundFetch ?

**Comment by hrishikesh-p** [comment]
Hey @pramod-practo , Do you see this happening still ?

**Comment by pramod636** [comment]
Hey @hrishikesh-p, Apologies for the delay.

Unfortunately, i cannot give you a better call stack than the one i gave previously. This is what we also got. 

The code which caused the crash in the background was 
```
let config = HotlineConfig.init(appID: Keys.kHotlineAppIDTest,
                                        andAppKey: Keys.kHotlineAppKeyTest)
```

We were not able to recreate this issue in our development environment and hence unfortunately, there is nothing i can help you with regards to the steps to recreate the issue.

But fortunately or unfortunately, we have stopped seeing crashes because of this in our prod environment for the past 3 - 4 days. So things are alright now.
